cordapp/src/main/kotlin/net/corda/plugins/SigningOptions.kt

```diff
@@ -5,28 +5,39 @@ import org.gradle.api.tasks.Input
 /** JAR sign options. */
 //Option used for ANT task "signjar"
 open class SigningOptions {
-    @get:Input
-    var alias = "cordapp-signer"
+    companion object {
+        // Defaults to keyStore resource/certificates/cordadevcakeys.jks with Corda developer (non-production) certificates
+        const val DEFAULT_ALIAS = "cordaintermediateca"
+        const val DEFAULT_STOREPASS = "cordacadevpass"
+        const val DEFAULT_KEYSTORE = "cordadevcakeys.jks"
+        const val DEFAULT_STORETYPE = "JKS"
+        const val DEFAULT_KEYPASS = "cordacadevkeypass"
+        // jarSigner doesn't add "SIG-" prefix, and EC signature doesn't follow JAR File Specification
```

Comment:
Can you elaborate on this? What happens if we use a non-standard signature file name? Do the standard Java APIs and tools still recognise it as a signed jar?

Reply:
Signature block file name in JAR file spec can be https://docs.oracle.com/javase/8/docs/technotes/guides/jar/jar.html
```
META-INF/*.DSA
META-INF/*.RSA
META-INF/SIG-*
```

and ` Digital Signatures` paragraph states: 
 ```
 The extension varies depending on the type of digital signature.
.RSA (PKCS7 signature, SHA-256 + RSA)
.DSA (PKCS7 signature, DSA)
Digital signature files for signature algorithms not listed 
above must reside in the META-INF directory and have the prefix "SIG-"
```

However JarSigner (https://docs.oracle.com/javase/8/docs/technotes/tools/windows/jarsigner.html) paragraph:  `The Signed JAR File` can produce (and it does):
```
A signature block file with a .DSA, .RSA, or .EC extension
```

So it produces invalid file name, and Corda internally checks for valid file names otherwise it throws exception (specifically it doesn't recognise that file).

Reply:
I check tested Java Security Manager (following this tutorial https://docs.oracle.com/javase/tutorial/security/toolsign/index.html)  using signature block file `META-INF/*.EC` and it is accepted. For `jarsigner` itself, if it also verifies as signed JARs with  `*.EC`. It's only the JAR file spec that states otherwise.  If we follow `jarsigner`way it's just matter to add `EC` to regex here https://github.com/corda/corda/blob/f81428eb53dcbc89462c052a94e893bb61e24749/core/src/main/kotlin/net/corda/core/internal/JarSignatureCollector.kt#L15